src/kudu/consensus/consensus_queue.h

```diff
@@ -564,8 +566,27 @@ class PeerMessageQueue {
                              ReplicaTypes replica_types,
                              const TrackedPeer* who_caused);
 
+  // Function to compute the commit index in flexi raft. Same as
+  // `AdvanceQueueWatermark` except that its only used for commit index
+  // advancement and takes in `data_commit_quorum` as a parameter which is a
+  // mapping from region to number of votes required in that region for a
+  // transaction to be considered as committed.
+  // Please note: `queue_lock_` is held as well as `lock_` from the associated
+  // RaftConsensus instance while this function gets called.
+  void AdvanceMajorityReplicatedWatermark(
```

Comment:
nit: Append "FlexRaft" or something similar to the name of this method so it's clear how it's different from the regular one.